### ISSUE TITLE
arange fails when `start` + `step` is greater than or equal to `stop`

### ISSUE THREAD

**Issue by piyueh** [nv-legate/legate.numpy#23]
## Problem

Something like `legate.numpy.arange(1, 3, 5)` thows an error `AttributeError: 'Future' object has no attribute 'compute_parallel_launch_space'`. (Here, `start` is `1`, `stop` is `3`, and `step` is  `5`. See the signature from [vanilla NumPy](https://numpy.org/doc/stable/reference/generated/numpy.arange.html).)

## To reproduce

1. step 1: create `test.py`:
   ```python
   from legate import numpy as legatenumpy
   import numpy as truenumpy
   t_start, t_end, dt = 0, 1, 2
   t_legate = legatenumpy.arange(t_start, t_end, dt)
   t_numpy = truenumpy.arange(t_start, t_end, dt)
   print(truenumpy.allclose(t_legate, t_numpy))
   print(t_legate)
   print(t_numpy)
   ```
2. step 2: run with `legate --cpus 1 ./test.py -lg:numpy:test`

## Expected result

Either
```
True
[0]
[0]
```
or `NotImplementedError`, or an error message indicating this specific use case of `arange` is not currently supported.

## Actual result

```python
Traceback (most recent call last):
  File "<prefix>/lib/python3.8/site-packages/legion_top.py", line 394, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "<prefix>/lib/python3.8/site-packages/legion_top.py", line 193, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "./test.py", line 6, in <module>
    t_legate = legatenumpy.arange(t_start, t_end, dt)
  File "<prefix>/lib/python3.8/site-packages/legate/numpy/module.py", line 61, in arange
    result._thunk.arange(start, stop, step, stacklevel=(stacklevel + 1))
  File "<prefix>/lib/python3.8/site-packages/legate/numpy/deferred.py", line 2338, in arange
    launch_space = dst.compute_parallel_launch_space()
AttributeError: 'Future' object has no attribute 'compute_parallel_launch_space'
```

## Notes

- Only throws the error when using `lg:numpy:test`.
- This use case of `arange` might be an edge case in most of applications. (I think)
- In production run (without `-lg:numpy:test`), I guess this error will probably never be triggered because this use case creates just-one-element array and does not go with legion codepath?

**Comment by lightsighter** [comment]
Fixed with `195cbbf0e8e`

**Comment by piyueh** [comment]
It works! Thanks. Closing the issue.
